File: app/src/main/java/com/lambdaschool/congressdetails/viewModel/DetailCongressPersonViewModel.java
package com.lambdaschool.congressdetails.viewModel;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.lambdaschool.congressdataapiaccess.CongresspersonDetails;
import com.lambdaschool.congressdetails.repo.DetailCongressPersonRepo;

public class DetailCongressPersonViewModel extends ViewModel {
    private MutableLiveData<CongresspersonDetails> liveData;

    public LiveData<CongresspersonDetails> getDetails(String id) {
        if (liveData == null) {
            loadData(id);
        }
        return liveData;
    }

    private void loadData(String id) {
        liveData = DetailCongressPersonRepo.getDetails(id);
    }
}

File: app/src/main/java/com/lambdaschool/congressdetails/viewModel/DetailCongressPersonViewModel.kt
package com.lambdaschool.congressdetails.viewModel

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import android.arch.lifecycle.ViewModel

import com.lambdaschool.congressdataapiaccess.CongresspersonDetails
import com.lambdaschool.congressdetails.repo.DetailCongressPersonRepo

class DetailCongressPersonViewModel : ViewModel() {
    private var liveData: MutableLiveData<CongresspersonDetails>? = null

    fun getDetails(id: String): LiveData<CongresspersonDetails>? {
        if (liveData == null) {
            loadData(id)
        }
        return liveData
    }

    private fun loadData(id: String) {
        liveData = DetailCongressPersonRepo.getDetails(id)
    }
}
